File: backend/sdk/ruby/lib/waas/services/webhooks.rb
# frozen_string_literal: true

module WAAS
  module Services
    class Webhooks
      def initialize(client)
        @client = client
      end

      def send(endpoint_id:, payload:, headers: nil)
        body = { endpoint_id: endpoint_id, payload: payload }
        body[:headers] = headers if headers

        response = @client.post("/webhooks/send", body)
        Models::SendWebhookResponse.from_hash(response)
      end
    end
  end
end

File: backend/sdk/php/src/Services/Webhooks.php
<?php

declare(strict_types=1);

namespace WAAS\Services;

use WAAS\Client;
use WAAS\Models\{SendWebhookRequest, SendWebhookResponse};

class Webhooks
{
    public function __construct(private readonly Client $client) {}

    public function send(SendWebhookRequest $request): SendWebhookResponse
    {
        $response = $this->client->post('/webhooks/send', $request->toArray());
        return SendWebhookResponse::fromArray($response);
    }
}
